### PR TITLE
fix dead lock

### PR DESCRIPTION
**Pull request by YarBor**

### CODE REVIEW

**Review by chickenlj — APPROVED** [review]
LGTM.

**Review by FoghostCn — APPROVED** [review]
LGTM

**Comment by FoghostCn** [comment]
pls add license for `router_test.go`

**Review by FoghostCn — CHANGES_REQUESTED** [review]
pls add license for `router_test.go`

File: cluster/router/script/router_test.go
Comment (FinalT) [inline cluster/router/script/router_test.go:1]: plz write unit test like other routers

File: cluster/router/script/router_test.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  }(invokers, invocation, context));
+`},
+			args: func() args {
+				res := args{}
+				res.invokers, res.invocation, _ = getRouteArgs()
+				return res
+			}(),
+			want: func(invokers []protocol.Invoker) bool {
+				return true
+			},
```
Comment (FinalT) [inline cluster/router/script/router_test.go:159]: why?

File: cluster/router/script/router_test.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  }(invokers,invocation,context));
+`},
+			args: func() args {
+				res := args{}
+				res.invokers, res.invocation, _ = getRouteArgs()
+				return res
+			}(),
+			want: func(invokers []protocol.Invoker) bool {
+				return true
+			},
```
Comment (FinalT) [inline cluster/router/script/router_test.go:131]: What's the point

**Review by FinalT — CHANGES_REQUESTED** [review]

File: cluster/router/script/router_test.go (hunk 2 above)
Comment (YarBor) [reply cluster/router/script/router_test.go:131]: expect to do nothing，the script will make panic, and recover, then ignore script route

**Pull request merged**